projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/alpha/BaseSuite.java

```diff
@@ -117,7 +118,15 @@ public void test() throws Exception {
             System.err.println(testDiscoveryPath);
             whiteListFilter = p -> !whiteList.contains(p) && p.startsWith(new File(suiteDir.toString(), testDiscoveryPath).toPath());
         }
-        return collectTestCases(suiteDir, whiteListFilter);
+        Collection<Object[]> testCases = collectTestCases(suiteDir, whiteListFilter);
+        if (testCases.size() != whiteList.size()) {
+            // we are missing test cases that are on the whitelist!
```

Comment:
This test harness only runs tests on the whitelist.
Also, it ensures that all tests on the whitelist pass.